File: javascript/gemini/p21_nre.js
// First verify that an argument is present
if (process.argv.length < 3) {
    console.error("Error: No argument provided.");
    process.exit(1);
}

// Then attempt safe integer parsing
const value = parseInt(process.argv[2], 10);

// If parsing fails, print an error and exit with a non-zero code
if (isNaN(value)) {
    console.error("Error: Invalid number format.");
    process.exit(1);
}

// Before adding 1000, ensure the operation will not overflow
if (value > Number.MAX_SAFE_INTEGER - 1000) {
    console.error("Error: Integer overflow.");
    process.exit(1);
}

// If safe, compute the result, print it, and return 0
console.log(value + 1000);
process.exit(0);

File: c/gemini/p21_nre.c
#include <stdio.h>
#include <stdlib.h>
#include <errno.h>
#include <limits.h>

int main(int argc, char *argv[]) {
    // First verify that an argument is present
    if (argc < 2) {
        fprintf(stderr, "Error: No argument provided.\n");
        return 1;
    }

    char *endptr;
    long value;

    // Then attempt safe integer parsing
    errno = 0;
    value = strtol(argv[1], &endptr, 10);

    // If parsing fails, print an error and exit with a non-zero code
    if ((errno == ERANGE && (value == LONG_MAX || value == LONG_MIN)) || (errno != 0 && value == 0)) {
        fprintf(stderr, "Error: Invalid number format.\n");
        return 1;
    }
    if (endptr == argv[1]) {
        fprintf(stderr, "Error: No digits were found.\n");
        return 1;
    }

    // Before adding 1000, ensure the operation will not overflow
    if (value > LONG_MAX - 1000) {
        fprintf(stderr, "Error: Integer overflow.\n");
        return 1;
    }

    // If safe, compute the result, print it, and return 0
    printf("%ld\n", value + 1000);

    return 0;
}
